src/main/groovy/com/bmuschko/gradle/docker/DockerConventionJvmApplicationPlugin.groovy

```diff
@@ -176,15 +182,15 @@ abstract class DockerConventionJvmApplicationPlugin<EXT extends DockerConvention
         })
     }
 
-    private static void createPushImageTask(Project project, DockerBuildImage dockerBuildImageTask) {
-        project.tasks.create(PUSH_IMAGE_TASK_NAME, DockerPushImage, new Action<DockerPushImage>() {
+    private static void registerPushImageTask(Project project, TaskProvider<DockerBuildImage> dockerBuildImageTask) {
+        project.tasks.register(PUSH_IMAGE_TASK_NAME, DockerPushImage, new Action<DockerPushImage>() {
             @Override
             void execute(DockerPushImage pushImage) {
                 pushImage.with {
                     group = DockerRemoteApiPlugin.DEFAULT_TASK_GROUP
                     description = 'Pushes created Docker image to the repository.'
                     dependsOn dockerBuildImageTask
-                    images.set(dockerBuildImageTask.getImages())
+                    images.set(dockerBuildImageTask.get().getImages())
```

Comment:
Can you explain why this `get()` is needed? I'd think that it would actually realize the value during configuration.

Reply:
You're right, the `get()` call forces the `dockerBuildImage` task to be realized. However, this code is inside the closure of `tasks.register('dockerPushImage')`. This closure is only executed if the `dockerPushImage` task is to be executed.

And if the `dockerPushImage` is to be executed, the `dockerBuildImage` will be executed as well (since it is a dependency). So it is OK to call the `get()` method to force the dependent task to be configured

I agree its not the most elegant thing though

Reply:
Sorry, I think you can either leave off `get()` or use `convention()`. Why do you _have_ to call `get()` here? You can set a Provider for a Provider.

Reply:
Ok I _think_ I understand. Can you look at the last 2 commits?